webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/SystemVariableService.scala

```diff
@@ -530,17 +533,50 @@ class SystemVariableServiceImpl(
     val varNodeGroups = systemVariableSpecService.get("RUDDER_NODE_GROUPS_VARS").toVariable(Seq(stringNodeGroupsVars))
     val varNodeGroupsClasses = systemVariableSpecService.get("RUDDER_NODE_GROUPS_CLASSES").toVariable(Seq(stringNodeGroupsClasses))
 
-    // If Syslog is disabled, we force HTTPS
-    val varNodeReportingProtocol = getSyslogProtocolDisabled() match {
-      case Full(true)  => systemVariableSpecService.get("REPORTING_PROTOCOL").toVariable(Seq(AgentReportingHTTPS.value))
-      case Full(false) => nodeInfo.nodeReportingConfiguration.agentReportingProtocol match {
-                            case Some(protocol) => systemVariableSpecService.get("REPORTING_PROTOCOL").toVariable(Seq(protocol.value))
-                            case _              => getProp("REPORTING_PROTOCOL", () =>  getReportProtocolDefault().map(_.value))
-                          }
-      case eb:EmptyBox =>
-        val e = eb ?~! s"Failed to get information on syslog protocol global status, fallbacking to default value."
-        logger.error( e.messageChain )
-        getProp("REPORTING_PROTOCOL", () => getReportProtocolDefault().map(_.value))
+    // If Syslog is disabled, we force HTTPS.
+    val varNodeReportingProtocol = {
+      // By default, we are tolerant: it's far worse to fail a generation while support is ok than to
+      // succeed and have a node that doesn't rerport.
+      // => version must exists and and starts by 2.x, 3.x, 4.x, 5.x
+      def versionHasHttps(maybeVersion: Option[AgentVersion]): Boolean = {
+        maybeVersion match {
+          case None    =>
+            true // we don't know, so say HTTPS is supported
+          case Some(v) =>
+            val forbiden = List("2.", "3.", "4.", "5.")
+            !forbiden.exists(x => v.value.startsWith(x))
+        }
+      }
+
+      def failure(nodeInfo: NodeInfo, badVersion: Option[AgentInfo], dsc: Option[AgentInfo]) = {
+        Failure(s"Node ${nodeInfo.hostname} (${nodeInfo.id.value}) has an agent which doesn't support sending compliance reports in HTTPS: ${
+                  dsc match {
```

Comment:
we should not blindly refuse dsc - the code that decides if it can send https reports or not is in the plugin - so we should at least limit to version known not to support HTTPS

Reply:
no version in 6.0 will suport it, and it's expected that all version in 6.1 support it. So I propose to let it like that for 6.0, and change it in 6.1 (but once we know it works) or think how to limit it at that time, knowing that the only correct solution is out os reach in 6.0 (because capabilities are not parsed).